src/helper.js

```diff
@@ -537,41 +542,37 @@ class helper {
 
 		let mosaicsFieldObject = [];
 
-		for (const resolvedMosaic of resolvedMosaics) {
-			if (resolvedMosaic.id.toHex() === http.networkCurrency.mosaicId) {
-				mosaicsFieldObject.push(this.networkCurrencyMosaicBuilder(resolvedMosaic));
-			} else {
-				if (0 < mosaicInfos.length && 0 < mosaicNames.length) {
-					let { divisibility } = mosaicInfos.find(info => info.mosaicId === resolvedMosaic.id.toHex());
-
-					mosaicsFieldObject.push({
-						...resolvedMosaic,
-						rawAmount: resolvedMosaic.amount,
-						mosaicId: resolvedMosaic.id.toHex(),
-						amount: helper.formatMosaicAmountWithDivisibility(resolvedMosaic.amount, divisibility),
-						mosaicAliasName: MosaicService.extractMosaicNamespace({ mosaicId: resolvedMosaic.id.toHex() }, mosaicNames)
-					});
-				}
-			}
-		}
+		const uniqueMosaicIds = [...new Set(resolvedMosaics.map(mosaic => mosaic.id.toHex()))];
```

Comment:
in line 525, why do you need to create new MosaicId, what is returned type of resolvedMosaic?

```
			const mosaicId = new MosaicId(resolvedMosaic.toHex()).id;
```

Reply:
in line 530, why are you filtering out network mosaic id?

Reply:
if you don't filter it out, uniqueMosaicIds would just be `[...new Set(resolvedMosaicIds)]`

Reply:
try to add test where 0 == resolvedMosaicIds.length

Reply:
> in line 525, why do you need to create a new MosaicId, what is returned type of resolvedMosaic?

it returns as `Id`, I will do some refactoring on this.

> in line 530, why are you filtering out network mosaic id?

those filtered mosaic ids need to call the API request mosaic info and namespace name, excluding network mosaic id, because we already have the information.